nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,326 @@
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(LOG_ANALYTICS_WORKSPACE_ID);
+        properties.add(LOG_ANALYTICS_CUSTOM_LOG_NAME);
+        properties.add(LINUX_PRIMARY_KEY);
+        properties.add(APPLICATION_ID);
+        properties.add(INSTANCE_ID);
+        properties.add(PROCESS_GROUP_IDS);
+        properties.add(JOB_NAME);
+        properties.add(SEND_JVM_METRICS);
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(final ReportingContext context) {
+        final String workspaceId = context.getProperty(LOG_ANALYTICS_WORKSPACE_ID).evaluateAttributeExpressions().getValue();
+        final String linuxPrimaryKey = context.getProperty(LINUX_PRIMARY_KEY).evaluateAttributeExpressions().getValue();
+        final boolean jvmMetricsCollected = context.getProperty(SEND_JVM_METRICS).asBoolean();
+
+        String logName = context.getProperty(LOG_ANALYTICS_CUSTOM_LOG_NAME).evaluateAttributeExpressions().getValue();
+        final String instanceId = context.getProperty(INSTANCE_ID).evaluateAttributeExpressions().getValue();
+        String groupIds = context.getProperty(PROCESS_GROUP_IDS).evaluateAttributeExpressions().getValue();
+        try {
+            if(groupIds == null) {
+                ProcessGroupStatus status = context.getEventAccess().getControllerStatus();
+                String processGroupName = status.getName();
+                List<Metric> allMetrics = collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected);
+                sendMetrics(workspaceId, linuxPrimaryKey, logName, allMetrics);
+            } else {
+                if (!groupIds.contains(",")) {
+                    ProcessGroupStatus status = context.getEventAccess().getGroupStatus(groupIds.trim());
+                    String processGroupName = status.getName();
+                    List<Metric> allMetrics = collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected);
+                    sendMetrics(workspaceId, linuxPrimaryKey, logName, allMetrics);
+                } else {
+                    List<Metric> allMetrics = new ArrayList<>();
+                    for(String groupId: groupIds.split(",")) {
+                        groupId =groupId.trim();
+                        ProcessGroupStatus status = context.getEventAccess().getGroupStatus(groupId);
+                        String processGroupName = status.getName();
+                        allMetrics.addAll(collectMetrics(instanceId, status, processGroupName, jvmMetricsCollected));
+                    }
+                    sendMetrics(workspaceId, linuxPrimaryKey, logName, allMetrics);
+                }
+            }
+        } catch (IOException | IllegalArgumentException e) {
+            getLogger().error("Exception in outmost block", e);
```

Comment:
Logging is fine, but throwing the exception away means we can't allow the NiFi framework to report and deal with this gracefully.

Reply:
Super onTrigger method from AbstractReportingTask,
public abstract void onTrigger(org.apache.nifi.reporting.ReportingContext arg0); does not allow me to rethrow exception. Besides, getLogger() returns the ComponentLog and the error shows in the UI. I think that this is the best we can do for now.

Reply:
Illegal argument is a runtime exception and I focused on that, so it needn't be declared in the parent -- but I checked the reporting task wrapper and it does less than I thought -- it only logs. I thought it administratively yielded, so I would agree...and any other unchecked RTE will be caught by the wrapper.